src/Fixme/Comment.hs

```diff
@@ -50,6 +53,19 @@ parseComments' filename =
     coalesce _ xs = xs
 
 
+-- | Read the given file, and parse out any comments.
+--
+-- Return Nothing if we cannot determine what language the file is in. Raises
+-- exceptions on bad IO, and also if the file cannot be decoded to Text.
+readComments :: FilePath -> IO (Maybe [Comment])
+readComments filename =
+  case languageForFile (toS filename) of
+    Nothing -> pure Nothing
+    Just language -> do
+      contents <- readFile filename
+      pure (Just $ parseComments (Just (toS filename)) language contents)
```

Comment:
(If you want to keep the original signature, replacing `<$>` with `` `traverse` `` in the above should do the trick, I think.)